src/jit/codegenxarch.cpp

```diff
@@ -7549,6 +7556,232 @@ void CodeGen::genSetRegToCond(regNumber dstReg, GenTreePtr tree)
     }
 }
 
+
+void CodeGen::genOverflowCheck_I8toI4(regNumber hiReg, regNumber loReg)
+{
+
+    // conv.ovf.i8.i4
+    // Generate the following sequence
+    // test loDWord, loDWord   // set flags
+    // jl neg
+    // pos: test hiDWord, hiDWord   // set flags
+    // jne ovf
+    // jmp done
+    // neg: cmp hiDWord, 0xFFFFFFFF
+    // jne ovf
+    // done:
```

Comment:
Interesting. I copied this from JIT32 source code :) Let me check if it can be simpler.